template_app/src/routes/+page.svelte

```diff
@@ -1,23 +1,78 @@
 <script lang="ts">
-	import { MapLibre } from 'svelte-maplibre';
-	import {SimpleComponent}  from 'ua-components';
-</script>
+  import {
+    Layout,
+    mapContents,
+    sidebarContents,
+  } from "ua-components/two_column_layout";
+  import { MapLibre } from "svelte-maplibre";
+  import TitleMode from "./TitleMode.svelte";
+  import TriangleMode from "./TriangleMode.svelte";
+  import { map as mapStore, backend, mode } from "./globals";
+  import workerWrapper from "./worker?worker";
+  import { type Backend } from "./worker";
+  import * as Comlink from "comlink";
+  import { onMount } from "svelte";
+  import type { Map } from "maplibre-gl";
+
+  // Everythign in this script section is boilerplate; you can ignore it
+
+  onMount(async () => {
+    // If you get "import declarations may only appear at top level of a
+    // module", then you need a newer browser.
+    // https://caniuse.com/mdn-api_worker_worker_ecmascript_modules
+    //
+    // In Firefox 112, go to about:config and enable dom.workers.modules.enabled
+    //
+    // Note this should work fine in older browsers when doing 'npm run build'.
+    // It's only a problem during local dev mode.
+    interface WorkerConstructor {
+      new (): Backend;
+    }
 
-<h1>Welcome to SvelteKit</h1>
-<p>Visit <a href="https://kit.svelte.dev">kit.svelte.dev</a> to read the documentation</p>
+    const MyWorker: Comlink.Remote<WorkerConstructor> = Comlink.wrap(
+      new workerWrapper(),
+    );
+    let backendWorker = await new MyWorker();
+    backend.set(backendWorker);
+  });
 
-<MapLibre style="https://basemaps.cartocdn.com/gl/positron-gl-style/style.json" standardControls />
-<div id='float'>
-	<SimpleComponent name="Stu" />
-</div>
+  let map: Map | undefined = undefined;
+  $: if (map) {
+    mapStore.set(map);
+  }
+
+  let sidebarDiv: HTMLDivElement;
+  let mapDiv: HTMLDivElement;
+  $: if (sidebarDiv && $sidebarContents) {
+    sidebarDiv.innerHTML = "";
+    sidebarDiv.appendChild($sidebarContents);
+  }
+  $: if (mapDiv && $mapContents) {
+    mapDiv.innerHTML = "";
+    mapDiv.appendChild($mapContents);
+  }
+</script>
 
+<Layout>
+  <div slot="left">
+    <h1>App title</h1>
+    <div bind:this={sidebarDiv}></div>
+  </div>
+  <div slot="main" style="position:relative; width: 100%; height: 100vh;">
+    <MapLibre
+      style="https://basemaps.cartocdn.com/gl/positron-gl-style/style.json"
+      standardControls
+      hash
+      bind:map
+    >
+      <div bind:this={mapDiv}></div>
 
-<style>
-	#float{
-		position:absolute;
-		top:20px;
-		right:20px;
-		background-color:white;
-		padding:20px;
-	}
-</style>
+      <!-- When you define new modes, you have to wire them up here -->
+      {#if $mode.kind == "title"}
+        <TitleMode />
+      {:else if $mode.kind == "triangle"}
+        <TriangleMode />
+      {/if}
+    </MapLibre>
+  </div>
```

Comment:
Modes here are different map states?

Reply:
ah ok they are different "pages" got it

Reply:
Yeah, I intend to explain this better in the user guide docs